pkg/dockerclient/dockerclient.go

```diff
@@ -208,7 +217,12 @@ func (c *Client) RunContainer(imageName string, runOptions *RunOptions) (string,
 
 // RemoveContainer removes a container given a container ID
 func (c *Client) RemoveContainer(containerID string) error {
-	_, err := c.cmdRunner.Run(nil, "docker rm -f %s", containerID)
+	cmd := "rm -f"
```

Comment:
I know this is the easiest place to put this... but I think this goes too far wrt breaking separation of concerns. This is a generic docker client - it does docker client stuff and I feel it should not know about this flag. In fact, this can be yanked to other projects or become a repo by itself if needed - this flag weighs it down.

A cleaner approach IMO is either to have a test docker client embedded this and overriding RemoveContainer / RemoveImage conditionally - also open to other things.

Reply:
I agree this is more sweeping than required however since we use it only for testing but IMO it's better than trying to find all the places in testing that deletes the image (which I failed at the beginning going with your approach).

Worse case we'll have some docker containers to clean manually.